### PR TITLE
Type: StopTime is in VehicleJourney without indirection

### PR DESCRIPTION
**Pull request by TeXitoi**
This is the most instanciated object, thus we optimize it.

A Journey computation is 5% faster with this modification, and memory
almost the same (a little bit less).

### CODE REVIEW

File: source/ed/tests/adapted_test.cpp [hunk 1]
```diff
@@ -2242,10 +2242,10 @@ BOOST_AUTO_TEST_CASE(impact_stoppoint_4){
     m.active_days = nt::Ven | nt::Lun;
     messages.push_back(m);
 
-    BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 2);
+    //BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 2);
     AtAdaptedLoader loader;
     loader.apply(messages, *b.data->pt_data);
-    BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 4);
+    //BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 4);
```
Comment (l-vincent-l) [inline source/ed/tests/adapted_test.cpp]: can you make a stop_times_size() function?

File: source/routing/dataraptor.h [hunk 2]
```diff
@@ -49,8 +49,8 @@ struct dataRAPTOR {
     std::vector<pair_int> footpath_index_backward;
     std::vector<uint32_t> arrival_times;
     std::vector<uint32_t> departure_times;
-    std::vector<type::StopTime*> st_idx_forward;// Nom a changer ce ne sont plus des idx mais des pointeurs
-    std::vector<type::StopTime*> st_idx_backward;
+    std::vector<const type::StopTime*> st_idx_forward;// Nom a changer ce ne sont plus des idx mais des pointeurs
```
Comment (antoine-de) [inline source/routing/dataraptor.h]: I think it's time to change the comment :wink:

Comment (antoine-de) [inline source/routing/dataraptor.h]: Ok, it is not nice to ask you that, but I think some more comment about st_idx_forward and st_idx_backward would be nice 

(and maybe some explanation about why the pointer to a vector contained object is not that bad in this case)

**Comment by antoine-de** [comment]
ok for me

File: source/routing/dataraptor.h [hunk 3]
```diff
@@ -47,10 +47,24 @@ struct dataRAPTOR {
     std::vector<pair_int> footpath_index_forward;
     std::vector<const navitia::type::StopPointConnection*> foot_path_backward;
     std::vector<pair_int> footpath_index_backward;
-    std::vector<uint32_t> arrival_times;
-    std::vector<uint32_t> departure_times;
-    std::vector<type::StopTime*> st_idx_forward;// Nom a changer ce ne sont plus des idx mais des pointeurs
-    std::vector<type::StopTime*> st_idx_backward;
+
+    // arrival_times (resp. departure_times) are the different arrival
+    // (resp. departure) times of each stop points sorted by
```
Comment (l-vincent-l) [inline source/routing/dataraptor.h]: stop time

**Pull request merged**